llvm/lib/Support/Windows/Program.inc

```diff
@@ -284,6 +285,8 @@ static bool Execute(ProcessInfo &PI, StringRef Program,
   unsigned CreateFlags = CREATE_UNICODE_ENVIRONMENT;
   if (AffinityMask)
     CreateFlags |= CREATE_SUSPENDED;
+  if (DetachProcess)
+    CreateFlags |= DETACHED_PROCESS;
```

Comment:
I don't know for Unix, but on Windows this flag prevents the I/O streams to be open on process startup, so any attempt to use `llvm::outs` or `llvm::errs` will most likely fail. Is that the intent?

Reply:
Another way for executing "detached" processes is to use `ShellExecuteEx`: https://learn.microsoft.com/en-us/windows/win32/shell/launch

Reply:
I would not say I am intentionally trying to prevent I/O streams from opening on process startup but the goal is to launch a process without a controlling terminal. When using `DETACHED_PROCESS` I do not expect `llvm::outs`, `llvm::errs`, or `stdin` to be useful until they have been redirected to a file (or another output location).

Regarding `ShellExecuteEx` is looks like I can have the process create it's own console but I don't see a way to execute a process without a console (https://learn.microsoft.com/en-us/windows/win32/api/shellapi/ns-shellapi-shellexecuteinfoa). I could be missing something.

Reply:
We discussed using `ShellExecute`, but it still gives you a console. The goal here is to not have a console at all. `DETACHED_PROCESS` gives the correct semantics for a daemon spawned by any given `clang` instance that doesn't belong to any specific instance.

Reply:
Understood, SGTM.